core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3235,4 +3235,17 @@ public boolean accept(CarbonFile file) {
     int version = fileHeader.getVersion();
     return ColumnarFormatVersion.valueOf((short)version);
   }
+
+  /**
+   * Check whether it is standard table means tablepath has Fact/Part0/Segment_ tail present with
+   * all carbon files. In other cases carbon files present directly under tablepath or
+   * tablepath/partition folder
+   * TODO Read segment file and corresponding index file to get the correct carbondata file instead
+   * of using this way.
+   * @param table
+   * @return
+   */
+  public static boolean isStandardCarbonTable(CarbonTable table) {
+    return !(table.isSupportFlatFolder() || table.isHivePartitionTable());
```

Comment:
Please add a check for sdk case also
CarbonTable.isTransactionalTable()

Reply:
This check is not valid here as some places they are explicitly checking it.

Reply:
ok